lib/rucio/core/transfer.py

```diff
@@ -1370,6 +1377,15 @@ def __list_transfer_requests_and_source_replicas(total_workers=0, worker_number=
                                          models.RSEFileAssociation.rse_id == models.Distance.src_rse_id)) \
         .with_hint(models.Distance, "+ index(distances DISTANCES_PK)", 'oracle')
 
+    # if transfertool specified, select only the requests where the source rses are set up for the transfer tool
+    if transfertool:
+        query = query.subquery()
+        subq = session.query(query) \
+            .join(models.RSEAttrAssociation, models.RSEAttrAssociation.rse_id == query.c.rse_id) \
+            .filter(models.RSEAttrAssociation.key == 'transfertool',
+                    models.RSEAttrAssociation.value == transfertool)
+        return subq.all()
```

Comment:
This skips the lower part, place in variable so that the lower part is executed as well.